### PR TITLE
feat: replace restricted-uploads.data with restricted-files.data

### LINKED ISSUE #2644: update file `restricted-upload.data`
There is no information on where this list comes from. We need to get some sources and update the list.

### PR DESCRIPTION
**Pull request by theseion**
restricted-uploads.data was a subset of restricted-files.data. restricted-files.data includes many more paths than restricted-uploads.data but is more restrictive.

Note that I chose to generate `restricted-upload.data` from `restricted-files.data` instead of using `restricted-files.data` directly. The reason is that the file name and path can be transmitted in separate fields. The `X-Filename` header, for example, will not contain the path but only the name.

Fixes #2644

### CODE REVIEW

File: rules/restricted-upload.data [hunk 1]
```diff
@@ -1,23 +1,212 @@
...
+.viminfo
+.vimrc
+.ws_ftp.ini
+.www_acl
+.wwwacl
+.xauthority
+.zhistory
+.zsh_history
+.zshrc
+0
```
[the remark this answers is not in the document]
Comment (theseion) [reply rules/restricted-upload.data:73]: Those are process IDs from `/proc/<ID>`. I'll remove those as they would very likely lead to many false positives.

File: rules/restricted-upload.data [hunk 2]
```diff
@@ -1,23 +1,212 @@
-# Apache webserver
+# This list can be generated from restricted-files.data by running the following shell command:
+# sed -i -e '7,$d' rules/restricted-upload.data && \
```
[the remark this answers is not in the document]
Comment (theseion) [reply rules/restricted-upload.data:2]: I want to retain the header and then append the output.

**Review by dune73 — CHANGES_REQUESTED** [review]
See comment in "conversation"

**Comment by theseion** [comment]
Good catch. I've extended the script to run the source list through spell.sh. This does not address all of the items in your list but at least the most obvious ones. Do you think that's enough?

**Comment by dune73** [comment]
Thanks.

We're now down to the following keywords, that might lead to FPs based on my previous assessment:

```
	Dockerfile
	asound
	config
	cpuinfo
	crypto
	filesystems
	hypervisor
	key-users
	meminfo
	misc
	zoneinfo
```

We might get away with most of these, but we can not add `crypto` and `misc` to a PL1 rule. `config` is a bit of an edge case for me. Using `crypto` in a filename should be legitimate, `misc` too of course. I think `config` falls into the same category: Uploading a file like `CNC-configuration.xxx` seems perfectly natural and should not lead to a FP, no matter it may resemble a critical application server configuration file that is being replaced during an ongoing attack.

I think we need to add at least these 3 terms to `./util/fp-finder/english-extended.txt` and remove them from this PR.

**Comment by theseion** [comment]
Done.

**Review by dune73 — APPROVED** [review]
OK, cool. Thanks.

This is a major expansion of the detection for rule 932180. This is potentially adding FPs too. Adding this here for the record.

Merging now.

**Pull request merged**